src/main/java/smartthings/ratpack/aws/internal/backoff/ExponentialBackoff.java

```diff
@@ -0,0 +1,32 @@
+package smartthings.ratpack.aws.internal.backoff;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ratpack.exec.Blocking;
+import ratpack.exec.Promise;
+
+/**
+ * Provides a mechanism for backoff inside a Ratpack promise chain.
+ */
+public class ExponentialBackoff {
+
+    private static final Logger log = LoggerFactory.getLogger(ExponentialBackoff.class);
+    private static final int[] FIBONACCI = new int[] { 1, 1, 2, 3, 5, 8, 13 };
+    private int attempts = 0;
+
+    public void mark() {
+        attempts = attempts >= FIBONACCI.length ? attempts : attempts + 1;
```

Comment:
Just a suggestion but maybe change this to an exponential function with a MAX value instead of a fixed array? 

```attempts = attempts >= MAX ? MAX : attempts + 1;
Thread.sleep(attempts * 2 * 1000)```

Also I believe the logic in `Thread.sleep(FIBONACCI[attempts] * 1000)` will create an index out of bounds exception when attempts is >= FIBONACCI.length.

Reply:
The index-out-of-bounds shouldn't happen because `attempts` is private and it is set to ensure it is within bounds.  That said, it was a bit weird doing a fibonacci algorithm in one labeled as an exponential based algorithm.  So I changed it as you suggested.

done.